File: activities/3-2/point.py
import math
class Point:
    def __init__(self, x, y):
        self.x = x
        self.y = y

    def print(self):
        print(f"({self.x}, {self.y})")

def distance(p1, p2):
    dx, dy = p1.x - p2.x, p1.y - p2.y
    return math.sqrt(dx*dx + dy*dy)

if __name__ == "__main__":
    p1 = Point(1, 2)
    p2 = Point(1, 3)
    p3 = Point(10, 12)

    p1.print()
    p2.print()
    p3.print()

    print("Distance between p1 and p3: ")
    print(distance(p1, p3))
    

File: homework09/paradigms/classes/Point.java
package paradigms.classes;

import java.util.Objects;
public class Point {
    private float x;
    private float y;

    public Point(float x, float y){
        this.x = x;
        this.y = y;
    }

    @Override
    public boolean equals(Object other){
        // reflexive
        if(other == this)
            return true;
        // non-null
        if(other == null)
            return false;
        // don't even bother! they have different types
        if(getClass() != other.getClass())
            return false;
        Point point = (Point) other; // why do we need this type cast?
        double p1_distance = Math.sqrt(this.x * this.x + this.y * this.y);
        double p2_distance = Math.sqrt(point.x * point.x + point.y * point.y);
        return p1_distance == p2_distance;
    }

    @Override
    public int hashCode(){
        return Objects.hash(this.x, this.y);
    }

    @Override
    public String toString() {
        return "[" + x + ", " + y + "]";
    }

}
